### ISSUE TITLE
[Bug]: Obsidian Git has issues with Credentials on startup after linking to second device

### ISSUE THREAD

**Issue by sublettj** [denolehov/obsidian-git#163]
### Describe the bug

When any of the git commands try to run in Obsidian, I first receive a GitHub login popup.  This login always fails with the same console message, copied below.

I followed foreveveryone's install procedure for windows 10 on my first machine, which was running fine and auto-backing up.  I then used almost the same installation procedure on a second machine.  Machine B is now running correctly, but Machine A has stopped connecting to GitHub.

Note, both machines are still logged into GitHub desktop, as the tutorial says, and I can push and pull from the desktop app.

![image](https://user-images.githubusercontent.com/71276779/148136443-b3e8a470-0581-41f5-9319-832971611fbd.png)
![image](https://user-images.githubusercontent.com/71276779/148136479-ad5b34e5-b4ac-4805-9c10-5f32f3e46671.png)

### Relevant errors (if available)

```shell
git obsidian error: Logon failed, use ctrl+c to cancel basic credential prompt.
bash: /dev/tty: No such device or address
error: failed to execute prompt script (exit code 1)
fatal: could not read Username for 'https://github.com': No such file or directory
Pushing to https://github.com/sublettj/Obsidian-Vaults.git
```

### Steps to reproduce

I'm not entirely sure.  I believe the issue started only after the second machine pushed its offline changes and I pulled those on machine A.

### Operating system

Windows

**Comment by Vinzent03** [comment]
Are you using your GitHub's account password for the password field or a [personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token)? Using the password doesn't work anymore.

**Comment by sublettj** [comment]
Hi Vinzent,
I have not used a PAT for either installations.  However, I just tried using one to see if it would resolve the issue I'm having on the first installation, and used the PAT in the password space on the github login popup that appears whenever I try to push on installation A.  That PAT as a password resulted in the same behavior as before, with the "login failed" obsidian popup.

I think I will try a reinstall process again to see if that gets it working.  The installation on my laptop is working wonderfully though.

**Comment by sublettj** [comment]
Ok, I've tried a few reinstallations, and I'm still having the weird login popup.  I don't know why I would be seeing that popup, since I'm using the "git config --global credential.helper wincred" command.  I tried resetting these credentials, but that didn't seem to help.

Sorry that I don't have more information to diagnose with.

Even starting from an installation on a new, empty repo I'm still having this issue.  I don't remember seeing the github login popup being mentioned anywhere in the windows install instructions.  The listed instructions also don't mention using a PAT, as they use the wincred.


I've compared the credentials between the two machines.

Machine A only has "Github - https://api.github.com/sublettj", while Machine B additionally has a "git:https://github.com" and "git:https://sublettj@github.com" credential.  I think this inconsistency may be the cause of the issues.

**Comment by sublettj** [comment]
Vinzent, I found the source of the issue.  Apparently, my GitHub for Windows was up-to-date, but my Git for Windows was not.  

Once I updated my git for windows, when I try to reauthenticate, the correct window appears.  This sign in with browser prompt creates the following git credentials that were missing on the other computer.  These credentials work and persist.
![image](https://user-images.githubusercontent.com/71276779/148463926-39249c3b-0351-4ea2-8d56-3736acac785e.png)
![image](https://user-images.githubusercontent.com/71276779/148464008-498e8347-561b-4e09-9fc5-7f1015a4ae6b.png)

For anyone with login issues, I'd start with making sure both the Github app and git for windows are up-to-date.


windows: git update-git-for-windows

Linux/Unix: git update

https://stackoverflow.com/questions/55768926/logon-failed-use-ctrlc-to-cancel-basic-credential-prompt-to-git-in-azure-devop
